src/Neo/SmartContract/Native/Notary.cs

```diff
@@ -0,0 +1,343 @@
+        {
+            long nFees = 0;
+            ECPoint[]? notaries = null;
+            foreach (var tx in engine.PersistingBlock.Transactions)
+            {
+                var attr = tx.GetAttribute<NotaryAssisted>();
+                if (attr is not null)
+                {
+                    notaries ??= GetNotaryNodes(engine.SnapshotCache);
+                    var nKeys = attr.NKeys;
+                    nFees += (long)nKeys + 1;
+                    if (tx.Sender == Hash)
+                    {
+                        var payer = tx.Signers[1];
+                        // Don't need to seal because Deposit is a fixed-sized interoperable, hence always can be serialized.
+                        var balance = engine.SnapshotCache.GetAndChange(CreateStorageKey(Prefix_Deposit, payer.Account))?.GetInteroperable<Deposit>();
+                        if (balance != null)
+                        {
+                            balance.Amount -= tx.SystemFee + tx.NetworkFee;
+                            if (balance.Amount.Sign == 0) RemoveDepositFor(engine.SnapshotCache, payer.Account);
+                        }
+                    }
+                }
+            }
+            if (nFees == 0) return;
+            if (notaries == null) return;
+            var singleReward = CalculateNotaryReward(engine.SnapshotCache, nFees, notaries.Length);
+            foreach (var notary in notaries) await GAS.Mint(engine, Contract.CreateSignatureRedeemScript(notary).ToScriptHash(), singleReward, false);
+        }
+
+        protected override void OnManifestCompose(IsHardforkEnabledDelegate hfChecker, uint blockHeight, ContractManifest manifest)
+        {
+            manifest.SupportedStandards = ["NEP-27"];
+        }
+
+        /// <summary>
+        /// Verify checks whether the transaction is signed by one of the notaries and
+        /// ensures whether deposited amount of GAS is enough to pay the actual sender's fee.
+        /// </summary>
+        /// <param name="engine">ApplicationEngine</param>
+        /// <param name="signature">Signature</param>
+        /// <returns>Whether transaction is valid.</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        private bool Verify(ApplicationEngine engine, byte[] signature)
+        {
+            if (signature is null || signature.Length != 64) return false;
+            var tx = engine.ScriptContainer as Transaction;
+            if (tx?.GetAttribute<NotaryAssisted>() is null) return false;
+            foreach (var signer in tx.Signers)
+            {
+                if (signer.Account == Hash)
+                {
+                    if (signer.Scopes != WitnessScope.None) return false;
+                    break;
+                }
+            }
+            if (tx.Sender == Hash)
+            {
+                if (tx.Signers.Length != 2) return false;
```

Comment:
!= or >2?

Reply:
`!=`. Exactly two signers is a requirement.